configuration_go/abstr/kubernetes/thanos/compactor/compactor.go

```diff
@@ -1 +1,269 @@
+	cmdopt.ExtraOpts
+}
+
+type CompactorStatefulSet struct {
+	Options    *CompactorOptions
+	VolumeType string
+	VolumeSize string
+
+	k8sutil.DeploymentGenericConfig
+}
+
+func (c *CompactorStatefulSet) WithLogLevel(level string) *CompactorStatefulSet {
+	c.Options.LogLevel = level
+	return c
+}
+
+// NewCompactor returns a new compactor statefulset with default values.
+// It allows generating the all the manifests for the compactor.
+func NewCompactor() *CompactorStatefulSet {
+	c := &CompactorOptions{
+		ObjstoreConfig:            "$(OBJSTORE_CONFIG)",
+		Wait:                      true,
+		LogLevel:                  "warn",
+		LogFormat:                 "logfmt",
+		DataDir:                   "/var/thanos/compactor",
+		RetentionResolutionRaw:    time.Hour * 24 * 365,
+		DeleteDelay:               time.Hour * 24 * 2,
+		CompactConcurrency:        1,
+		DownsampleConcurrency:     1,
+		DeduplicationReplicaLabel: "replica",
+	}
+
+	commonLabels := map[string]string{
+		k8sutil.NameLabel:      "thanos-compact",
+		k8sutil.InstanceLabel:  "observatorium",
+		k8sutil.PartOfLabel:    "observatorium",
+		k8sutil.ComponentLabel: "database-compactor",
+	}
+
+	labelSelectors := map[string]string{
+		k8sutil.NameLabel:     commonLabels[k8sutil.NameLabel],
+		k8sutil.InstanceLabel: commonLabels[k8sutil.InstanceLabel],
+	}
+
+	namespaces := []string{defaultNamespace}
+
+	return &CompactorStatefulSet{
+		Options: c,
+		DeploymentGenericConfig: k8sutil.DeploymentGenericConfig{
+			Image:                defaultImage,
+			ImageTag:             defaultImageTag,
+			ImagePullPolicy:      corev1.PullIfNotPresent,
+			Name:                 "observatorium-thanos-compact",
+			Namespace:            defaultNamespace,
+			CommonLabels:         commonLabels,
+			Replicas:             1,
+			PodResources:         k8sutil.NewResourcesRequirements("2", "3", "2000Mi", "3000Mi"),
+			Affinity:             *k8sutil.NewAntiAffinity(namespaces, labelSelectors),
+			SecurityContext:      k8sutil.GetDefaultSecurityContext(),
+			EnableServiceMonitor: true,
+			LivenessProbe: k8sutil.NewProbe("/-/healthy", defaultHTTPPort, k8sutil.ProbeConfig{
+				FailureThreshold: 4,
+				PeriodSeconds:    30,
+			}),
+			ReadinessProbe: k8sutil.NewProbe("/-/ready", defaultHTTPPort, k8sutil.ProbeConfig{
+				FailureThreshold: 20,
+				PeriodSeconds:    5,
+			}),
+			TerminationGracePeriodSeconds: 120,
+			ConfigMaps:                    make(map[string]map[string]string),
+			Env: []corev1.EnvVar{
+				k8sutil.NewEnvFromSecret("OBJSTORE_CONFIG", "objectStore-secret", "thanos.yaml"),
+				k8sutil.NewEnvFromField("HOST_IP_ADDRESS", "status.hostIP"),
+			},
+		},
+		VolumeType: "gp2-csi",
+		VolumeSize: "50Gi",
+	}
+}
+
+// Manifests returns the manifests for the compactor.
+// It includes the statefulset, the service, the service monitor, the service account and the config maps required by the containers.
+func (c *CompactorStatefulSet) Manifests() k8sutil.ObjectMap {
```

Comment:
Why do we treat Secrets differently here? The output here depends on a Secret and it seems fairly rigid in that regards. Is it out of scope here for some reason?

Reply:
This is a mistake. I have added capability for associating secrets to containers. The same way this is done for configMaps.